Tests/FluentKitTests/Utility/DummyDatabaseForTestQueries.swift

```diff
@@ -0,0 +1,97 @@
+@testable import FluentKit
+import NIO
+import SQLKit
+
+/// Lets you determine the row results for each query.
+///
+/// Initialize with an array or arrays, each inner array
+/// containing all the rows for a certain query result. Each
+/// new query will produce the next inner-array of results.
+///
+/// You can specify an empty result-set (as is correct for
+/// some queries and useful for other tests) with an empty
+/// inner-array.
+///
+/// Examples:
+///
+/// Return an empty result for all queries:
+///
+///     DummyDatabaseForTestQueries(mockResults: [[]])
+///
+/// Return an empty result for first query, and a single result
+/// for the second query (perhaps a query to find a record with
+/// no results followed by a successful query to create the record):
+///
+///     DummyDatabaseForTestQueries(mockResults: [
+///         [],
+///         [
+///             DummyDatabaseForTestQueries.DummyRow(dummyDecodedFields: ["id": 1, "name": "Boise"])
+///         ]
+///     ])
+public class DummyDatabaseForTestQueries: Database {
```

Comment:
Maybe we could call this just `TestDatabase` and continue to build on useful testing functionality?

Reply:
Also maybe this would be useful in FluentKit itself so that the benchmarker or people writing tests for their app could use it.

Reply:
> Maybe we could call this just TestDatabase and continue to build on useful testing functionality?

Intriguing idea. My initial reaction was "yeah but the other current test database only needs to serialize SQL which is pretty different than needing to mock responses to those queries." _But_ there is definitely a path forward for this to serve both purposes, I guess it just depends on how robust we want the **one true test database** to be when decomposing the different use-cases into their own test database classes could prove a bit more straight forward.

I'm not opposed, just wondering if there is a valuable enough trade-off there.

> Also maybe this would be useful in FluentKit itself so that the benchmarker or people writing tests for their app could use it.

The current implementation seems a bit low-level at the moment to provide much general value to ppl testing their vapor apps, but I again certainly see how this could be wrapped in something higher level (I imagine it being more useful for vapor app authors to mock the `Model` being returned than the fields of the database rows). Want me to move this into a new `FluentKitTesting` framework that gets pulled into the `FluentKitTests` target? That seems a bit more idiomatic than throwing it into the `FluentKit` target.

Reply:
> Intriguing idea. My initial reaction was "yeah but the other current test database only needs to serialize SQL which is pretty different than needing to mock responses to those queries."

That one seems more specific to me and could possibly be renamed to `TestSQLDatabase`. The one added in this PR seems more generally useful for testing, though maybe we could come up with a name like `TestResultDatabase` or something. 

> Want me to move this into a new FluentKitTesting framework that gets pulled into the FluentKitTests target?

Yeah I like the idea of a new target. Could we copy what Vapor does and call it `XCTFluent` (Xcode Test Fluent): https://github.com/vapor/vapor/blob/master/Package.swift#L87

Reply:
- Moved into `XCTFluent` library.
- Renamed to `TestDatabase` (I figure it can be renamed to `TtestResultDatabase` or similar later if it becomes meaningful to distinguish it from other offerings in `XCTFluent`)
- Left the SQL serializer test DB alone, mostly. No strong opinions but it does not immediately seem all that useful to test targets external to FluentKit.